### ISSUE TITLE
Loss of significant error messages

### ISSUE THREAD

**Issue by stavro** [ZennerIoT/ex_audit#1]
Wrapping all calls to insert, update, etc causes valuable information to be lost about normal database transactions.

With a schema that uses [Changeset.foreign_key_constraint/3](https://hexdocs.pm/ecto/Ecto.Changeset.html#foreign_key_constraint/3) or [Changeset.no_assoc_constraint](https://hexdocs.pm/ecto/Ecto.Changeset.html#no_assoc_constraint/3), the messages are lost.

```elixir
def prepare_delete(category) do
  category
  |> change
  |> no_assoc_constraint(:topics)
end



# Without using ExAudit.Repo

Category
|> Repo.get(id)
|> Category.prepare_delete
|> Repo.delete

# {:error,
#  #Ecto.Changeset<action: :delete, changes: %{},
#   errors: [topics: {"are still associated with this entry", []}],
#   data: #Category<>, valid?: false>}


# Using ExAudit.Repo

Category
|> Repo.get(id)
|> Category.prepare_delete
|> Repo.delete

# {:error, :rollback}
``` 

I realize this is not a straightforward (or even possible) thing to fix, but it should be very well documented that database errors and constraint messages are lost.  I have a few places where there are multiple database-enforced constraints, and when using the provided Repo module, I have no way of knowing which constraint failed

**Comment by narrowtux** [comment]
Seems like Repo.transaction will return `{:error, :rollback}` all the time, I thought returning the original return value of the wrapped repo function would suffice. I'm looking into alternatives, since I really want everything to work like before.

**Comment by narrowtux** [comment]
In the hopes that Ecto.Multi will not lose the error messages, I rewrote my transaction code to use a multi. Now I've just got to write a test

**Comment by narrowtux** [comment]
Fixed in 0.3.1 @stavro
